docs/migration-v8.md

```diff
@@ -1 +1,515 @@
+Policy
+  .Handle<SomeExceptionType>()
+  .WaitAndRetry(3, _ => TimeSpan.FromSeconds(1));
+
+// Wait and retry multiple times with callback
+Policy
+    .Handle<SomeExceptionType>()
+    .WaitAndRetry(3, _ => TimeSpan.FromSeconds(1), onRetry: (exception, retryCount) =>
+    {
+        // Add logic to be executed before each retry, such as logging
+    });
+
+// Wait and retry forever
+Policy
+    .Handle<SomeExceptionType>()
+    .WaitAndRetryForever(_ => TimeSpan.FromSeconds(1));
+```
+<!-- endSnippet -->
+
+### Retry and wait in v8
+
+<!-- snippet: migration-retry-wait-v8 -->
+```cs
+// Wait and retry multiple times
+new ResiliencePipelineBuilder().AddRetry(new RetryStrategyOptions
+{
+    ShouldHandle = new PredicateBuilder().Handle<SomeExceptionType>(),
+    MaxRetryAttempts = 3,
+    Delay = TimeSpan.FromSeconds(1),
+    BackoffType = DelayBackoffType.Constant
+})
+.Build();
+
+// Wait and retry multiple times with callback
+new ResiliencePipelineBuilder().AddRetry(new RetryStrategyOptions
+{
+    ShouldHandle = new PredicateBuilder().Handle<SomeExceptionType>(),
+    MaxRetryAttempts = 3,
+    Delay = TimeSpan.FromSeconds(1),
+    BackoffType = DelayBackoffType.Constant,
+    OnRetry = args =>
+    {
+        // Add logic to be executed before each retry, such as logging
+        return default;
+    }
+})
+.Build();
+
+// Wait and retry forever
+new ResiliencePipelineBuilder().AddRetry(new RetryStrategyOptions
+{
+    ShouldHandle = new PredicateBuilder().Handle<SomeExceptionType>(),
+    MaxRetryAttempts = int.MaxValue,
+    Delay = TimeSpan.FromSeconds(1),
+    BackoffType = DelayBackoffType.Constant
+})
+.Build();
+```
+<!-- endSnippet -->
+
+- It's important to remember that the configuration in v8 is options based, i.e. `RetryStrategyOptions` are used.
+- Notice that to disable waiting between retries the `Delay` property is set to `TimeSpan.Zero`.
+- To retry forever set `MaxRetryAttempts` to `int.MaxValue`.
+
+## Migrating bulkhead policy
+
+The bulkhead policy is now replaced by the [rate limiter strategy](strategies/rate-limiter.md) which uses the [`System.Threading.RateLimiting`](https://www.nuget.org/packages/System.Threading.RateLimiting) package. The new counterpart to bulkhead is `ConcurrencyLimiter`.
+
+### Bulkhead in v7
+
+<!-- snippet: migration-bulkhead-v7 -->
+```cs
+// Create sync bulkhead
+ISyncPolicy syncPolicy = Policy.Bulkhead(maxParallelization: 100, maxQueuingActions: 50);
+
+// Create async bulkhead
+IAsyncPolicy asyncPolicy = Policy.BulkheadAsync(maxParallelization: 100, maxQueuingActions: 50);
+
+// Create generic sync bulkhead
+ISyncPolicy<HttpResponseMessage> syncPolicyT = Policy.Bulkhead<HttpResponseMessage>(maxParallelization: 100, maxQueuingActions: 50);
+
+// Create generic async bulkhead
+IAsyncPolicy<HttpResponseMessage> asyncPolicyT = Policy.BulkheadAsync<HttpResponseMessage>(maxParallelization: 100, maxQueuingActions: 50);
+```
+<!-- endSnippet -->
+
+### Concurrency limiter in v8
```

Comment:
I am not 100% sure, but it might be a good idea just for completeness put Bulkhead in the title as well, in case people don't read and just go by headers, so that it's immediately obvious they belong together.